ftw/simplelayout/contenttypes/contents/textblock.py

```diff
@@ -3,19 +3,25 @@
 from ftw.simplelayout import _
 from ftw.simplelayout.browser.actions import DefaultActions
 from ftw.simplelayout.contenttypes.contents.interfaces import ITextBlock
+from ftw.simplelayout.images.interfaces import IImageLimits
+from ftw.simplelayout.images.validators import ImageLimitValidator
 from ftw.simplelayout.interfaces import IBlockConfiguration
 from ftw.simplelayout.interfaces import IBlockModifier
 from plone.app.textfield import RichText
 from plone.autoform.interfaces import IFormFieldProvider
 from plone.dexterity.content import Item
 from plone.directives import form
 from plone.namedfile.field import NamedBlobImage
+from z3c.form import validator
 from zope import schema
 from zope.i18n import translate
 from zope.interface import alsoProvides
 from zope.interface import implements
 
 
+TEXTBLOCK_IMAGE_LIMIT_IDENTIFIER = 'textblock'
```

Comment:
Could you update me on your thoughts here? Did you consider using the portal_type? Why did you discard that?

Reply:
Do you mean:

`TEXTBLOCK_IMAGE_LIMIT_IDENTIFIER = 'ftw.simplelayout.TextBlock'`?

That would be ok.

My first implementation was more generic. It was possible to set the config identifier like this:

`ftw.simplelayout.contenttypes.contents.textblock.ITextBlock.image`

and it was not necessary to define your own identifier. And why this long, complicated identifier? The Add-Form only provides the schema interface and a schema can have multiple image fields. 

But I didn't like it because it would have to implement a lot of adapters to make it more or less configurable if you want to customize it. Also I had to implement a lot of ugly lookups (i.e. the schema field from a contexts schema-interface and so on (I still have this branch if you are interested to see the previous implementation). Even the functions were lot complicated because this dynamic identifier.

The new implementation with the static identifier is much easier to understand, to configure and to customize.

Reply:
I just run into problems with my implementation.

Using the portal-type would be the best. Within the add-form I could use the portal-type hardcoded. Or do you have a hint on how to properly get the portal-type within the validator?

If I do not need to differ the limits for different image-fields, this would be ⭐️ (and I think there was no such requirement 😉 )

Reply:
Yes, I would also go for the portal_type. Because it is simpler to understand and it should be good enough. We could also go with the fieldname `image` hard-coded (not sure about graphic block). So 👍